### ISSUE TITLE
pre-move deadlock

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#812]
Original [issue 812](https://code.google.com/p/pychess/issues/detail?id=812) reported by [mattgatto](https://code.google.com/u/mattgatto/) 2013-08-25

I was playing a fischer random game on FICS and while waiting for my opponent to move, I made a pre-move capture, PxNe4, after which my opponent's move, Bh7 came through and then an immediate lockup occurred. This is end of the log:

16:11:05 Default Debug: GameModel.run: releasing self.applyingMoveLock
16:11:05 Default Debug: GameModel.run: id=139975616647888, players=[mgatto (1665), GuestVQLS (U)], self.ply=16: updating mgatto (1665)'s time
16:11:05 Default Debug: GameModel.run: id=139975616647888, players=[mgatto (1665), GuestVQLS (U)], self.ply=16: calling mgatto (1665).makeMove()
16:11:05 Default Debug: Human.makeMove: move=g8h7, board1=#White FAfa -
# ♜ ♝ ♛ ♚ . ♜ . .
# ♟ ♟ . . . ♟ ♟ ♝
# . . . ♞ ♟ . ♞ ♟
# . . ♟ . . . . .
# ♙ . . . ♘ ♙ . .
# . . . ♙ . ♘ ♙ .
# . ♙ ♙ . . . . ♙
# ♖ ♗ ♕ ♔ . ♖ ♗ .
# board2=#Black FAfa -
# ♜ ♝ ♛ ♚ . ♜ ♝ .
# ♟ ♟ . . . ♟ ♟ .
# . . . ♞ ♟ . ♞ ♟
# . . ♟ . . . . .
# ♙ . . . ♘ ♙ . .
# . . . ♙ . ♘ ♙ .
# . ♙ ♙ . . . . ♙
# ♖ ♗ ♕ ♔ . ♖ ♗ .
# 

16:11:07 ('GuestYDNR', 'raw') Debug: 
<wa> Pfehlauer 001029 0P0P1658P0P0P0P0P0P
fics% 
Pfehlauer Blitz (1029), Std (----), Wild (1658), Light(----), Bug(----)
  is now available for matches.
fics% 
16:11:07 ('GuestYDNR', 'nonmatched') Debug: 
16:11:07 ('GuestYDNR', 'lines') Debug: <wa> Pfehlauer 001029 0P0P1658P0P0P0P0P0P

**Comment by pychess** [comment]
- attached [pychess.log.tar.bz2](https://code.google.com/p/pychess/issues/detail?id=812#c0)

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=812#c1) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2013-08-27

Reverting that last change gets rid of the deadlock:

[gatto@fedora-19 pychess]$ hg diff
diff -r a904a3c874d4 lib/pychess/widgets/BoardView.py
--- a/lib/pychess/widgets/BoardView.py  Wed Aug 14 05:47:46 2013 -0700
+++ b/lib/pychess/widgets/BoardView.py  Tue Aug 27 05:55:44 2013 -0700
@@ -1386,9 +1386,9 @@
             self.showPrev()

```
 def setPremove(self, premovePiece, premove0, premove1, premovePly, promotion=None):
```
-        with self.animationLock:
-            self.premovePiece = premovePiece
-            self.premove0 = premove0
-            self.premove1 = premove1
-            self.premovePly = premovePly
-            self.premovePromotion = promotion
  +#        with self.animationLock:
-        self.premovePiece = premovePiece
-        self.premove0 = premove0
-        self.premove1 = premove1
-        self.premovePly = premovePly
-        self.premovePromotion = promotion

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=812#c2) posted by [gbtami](https://code.google.com/u/gbtami/) 2013-08-27

This issue was closed by revision f3c870323931.
- changed status to: Fixed

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=812#c3) posted by [gbtami](https://code.google.com/u/gbtami/) 2013-08-27

Thx, Matt!
